packages/sanes-chrome-extension/src/extension/background/messages/index.ts

```diff
@@ -98,9 +101,12 @@ export interface MessageToForeground {
   readonly data?: any;
 }
 
-export function isMessageToForeground(data: unknown): data is MessageToForeground {
+export function isMessageToForeground(data: unknown, expectedAction: string): data is MessageToForeground {
```

Comment:
Please don't extend this type guard function. This should only check the type at runtime to allow type narrowing. This is the reason for the special return type `data is MessageToForeground`. See "User Defined Type Guards" in https://basarat.gitbooks.io/typescript/docs/types/typeGuard.html#user-defined-type-guards

What you could do instead is

```
const transactionsChangesMessage = isMessageToForeground(msg) && msg.action === MessageToForegroundAction.TransactionsChanges);
                                                                 ^^^ here TS knows what type msg is
```